### PR TITLE
Removed Creator field in ModelVersionView if user_id is undefined

### PR DESCRIPTION
**Pull request by jinzhang21**
Signed-off-by: Jin Zhang <jin.zhang@databricks.com>

## What changes are proposed in this pull request?
Removed creator field on the model version view page, if user_id is not defined.

## How is this patch tested?

Unit test

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

## Release Notes
Fixed an issue that creator field shows empty on the model version view page.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [x] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [x] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/server/js/src/model-registry/components/ModelVersionView.js [hunk 1]
```diff
@@ -214,14 +214,16 @@ export class ModelVersionViewImpl extends React.Component {
...
     return (
-      <Descriptions.Item
-        label={this.props.intl.formatMessage({
-          defaultMessage: 'Creator',
-          description: 'Label name for creator metadata in model version page',
-        })}
-      >
-        {user_id}
-      </Descriptions.Item>
+      user_id && (
```
Comment (sunishsheth2009) [inline mlflow/server/js/src/model-registry/components/ModelVersionView.js:217]: The code usage
```
    const defaultOrder = [
      this.renderRegisteredTimestampDescription(modelVersion.creation_timestamp),
      this.renderCreatorDescription(modelVersion.user_id),
      this.renderStageDropdown(modelVersion),
      this.renderLastModifiedDescription(modelVersion.last_updated_timestamp),
      this.renderSourceRunDescription(),
    ];
 ```
 
 I think the entry in the array here would be undefined, not sure if that is ok to do?
 
 We can also do
 ```
 let defaultOrder = [
      this.renderRegisteredTimestampDescription(modelVersion.creation_timestamp),
      this.renderStageDropdown(modelVersion),
      this.renderLastModifiedDescription(modelVersion.last_updated_timestamp),
      this.renderSourceRunDescription(),
 ];
 
 if(user_id) {
          defaultOrder = [...defaultOrder, this.renderCreatorDescription(modelVersion.user_id),]
 }
 ```
 Not sure what the intended behavior is.

Comment (jinzhang21) [reply mlflow/server/js/src/model-registry/components/ModelVersionView.js:217]: I had the same doubt. The unit test was testing this exact behavior. The current code seems to be working fine. Your suggested approach would also work but the order of rendering will deviate from its current behavior. Of course I can make it a joint of 2 or 3 lists to maintain that ordering. What's the preferred JS style in this case?

Comment (sunishsheth2009) [reply mlflow/server/js/src/model-registry/components/ModelVersionView.js:217]: Ahhh that makes sense to maintain ordering. So we can do something like
```
defaultOrder = [... defaultOrder.slice(0, 1), this.renderCreatorDescription(modelVersion.user_id), ...defaultOrder.slice(2)];
```

Comment (jinzhang21) [reply mlflow/server/js/src/model-registry/components/ModelVersionView.js:217]: I did this instead:
` return defaultOrder;`
to
` return defaultOrder.filter(x => x !== null);`
Does it work?

Comment (sunishsheth2009) [reply mlflow/server/js/src/model-registry/components/ModelVersionView.js:217]: Yes I think that should work as well

**Review by sunishsheth2009 — APPROVED** [review]

**Pull request merged**